WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/AddEditCoupon.swift

```diff
@@ -15,29 +15,196 @@ struct AddEditCoupon: View {
+                                                 selectionStyle: .disclosure, action: { })
+                                Divider()
+                                    .padding(.leading, Constants.margin)
+                            }
+
+                            Group {
+                                TitleAndToggleRow(title: Localization.includeFreeShipping, isOn: .constant(false))
+                                    .padding(.horizontal, Constants.margin)
+                                Divider()
+                                    .padding(.leading, Constants.margin)
+                            }
+                        }
+
+                        Group {
+                            ListHeaderView(text: Localization.headerApplyCouponTo.uppercased(), alignment: .left)
+
+                            // TODO: add a new style with the icon on the left side
+                            Button {
+                                //TODO: handle action
+                            } label: {
+                                Text(Localization.editProductsButton)
+                                    .bodyStyle()
+                            }
+                            .buttonStyle(SecondaryButtonStyle())
+                            .padding(.horizontal, Constants.margin)
+
+                            // TODO: add a new style with the icon on the left side
+                            Button {
+                                //TODO: handle action
+                            } label: {
+                                Text(Localization.editProductCategoriesButton)
+                                    .bodyStyle()
+                            }
+                            .buttonStyle(SecondaryButtonStyle())
+                            .padding(.horizontal, Constants.margin)
+                        }
+
+                        Group {
+                            ListHeaderView(text: Localization.headerUsageDetails.uppercased(), alignment: .left)
+
+                            TitleAndValueRow(title: Localization.usageRestrictions,
+                                             value: .placeholder(""),
+                                             selectionStyle: .disclosure, action: { })
+                            Divider()
+                                .padding(.leading, Constants.margin)
+                        }
+
+                        Button {
+                            //TODO: handle action
+                        } label: {
+                            Text(Localization.saveButton)
+                        }
+                        .buttonStyle(PrimaryButtonStyle())
+                        .padding(.horizontal, Constants.margin)
+                        .padding(.top, Constants.verticalSpacing)
+                    }
+                }
+                .ignoresSafeArea(.container, edges: [.horizontal, .bottom])
```

Comment:
I think maybe we should keep the content above the bottom safe area? Just so the action button at the bottom will not be too close to the bottom edge of the screen:

<img src="https://user-images.githubusercontent.com/5533851/161526593-914a28fc-3198-4ead-9be5-d92512a84c75.png" width=320 />